packages/sdk/src/chains/BaseAssetTransfer.ts

```diff
@@ -72,6 +73,26 @@ export abstract class BaseAssetTransfer {
     return transfer;
   }
 
+  /**
+   * Returns all registered resources between source domain of asset transfer and provided destination domain (on-chain check).
+   *
+   * @param {string} destinationDomainID - Destination domain ID
+   */
+  public async getRegisteredResourcesTo(destinationDomainID: string): Promise<Resource[]> {
+    const registeredResources: Resource[] = [];
+    if (this.config.getSourceDomainConfig().id.toString() === destinationDomainID) {
+      throw new Error('Provided destination domain same as source domain');
+    }
+    for (const resource of this.config.getSourceDomainConfig().resources) {
+      if (await this.isRouteRegistered(destinationDomainID, resource)) {
```

Comment:
I guess each call makes rpc call. Maybe it would make sense to use some multicall read contract to fetch all those in single rpc call but considering we will replace with json endpoint this can do form now!